### ISSUE TITLE
parsing of platforms_xxxx.conf does not catch all missing variables, error message not very descriptive

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#207]
In GitLab by @etiennesky on Nov 9, 2016, 11:42

When creating a new experiment I made a small mistake in the MN3 platform description, SCRATCH_DIR was wrongly named CRATCH_DIR after editing.

"autosubmit create" ran fine, but "autosubmit run" gave a cryptic error "AttributeError: 'NoneType' object has no attribute 'endswith'"

First, this missing variable should probably be reported (as I think other variables such as USER are checked), and there should be a more user-friendly error reported.

```
etourign@ bscesautosubmit01:/esnas/autosubmit/a0ch/tmp> autosubmit run a0ch
Preparing .lock file to avoid multiple instances with same expid.

Checking configuration files...
autosubmit_a0ch.conf OK
platforms_a0ch.conf OK
jobs_a0ch.conf OK
expdef_a0ch.conf  OK
Configuration files OK

[CRITICAL] Unhandled exception on Autosubmit: 'NoneType' object has no attribute 'endswith'
Traceback (most recent call last):
  File "/shared/earth/software/autosubmit/3.7.5-foss-2015a-Python-2.7.9/lib/python2.7/site-packages/autosubmit-3.7.5-py2.7.egg/autosubmit/autosubmit.py", line 324, in parse_args
    return Autosubmit.run_experiment(args.expid)
  File "/shared/earth/software/autosubmit/3.7.5-foss-2015a-Python-2.7.9/lib/python2.7/site-packages/autosubmit-3.7.5-py2.7.egg/autosubmit/autosubmit.py", line 602, in run_experiment
    submitter.load_platforms(as_conf)
  File "/shared/earth/software/autosubmit/3.7.5-foss-2015a-Python-2.7.9/lib/python2.7/site-packages/autosubmit-3.7.5-py2.7.egg/autosubmit/platforms/paramiko_submitter.py", line 129, in load_platforms
    remote_platform.user, remote_platform.expid)
  File "/shared/earth/software/Python/2.7.9-foss-2015a/lib/python2.7/posixpath.py", line 77, in join
    elif path == '' or path.endswith('/'):
AttributeError: 'NoneType' object has no attribute 'endswith'
```

**Comment by kinow** [comment]
In GitLab by @ jlope2 on Nov 9, 2016, 12:18

Hi @etiennesky,

I don't know why but this variable is not among those that are reviewed.

I proceed to add it.

Thanks!

**Comment by kinow** [comment]
In GitLab by @ jlope2 on Nov 9, 2016, 15:35

@etiennesky do you think there are more variables that should be included in the ```platforms.conf``` checking?

The variables that are checked now are:

*  TYPE
*  PROJECT
*  USER
*  HOST
*  SCRATCH_DIR (soon with the version 3.7.6)

Thanks

**Comment by kinow** [comment]
In GitLab by @etiennesky on Nov 9, 2016, 15:38

Thanks @ jlope2  that looks sufficient.

However, is there a way to make sure that the error thrown is more descriptive in similar situations? In this case I had no indication as to which variable was missing.

**Comment by kinow** [comment]
In GitLab by @ jlope2 on Nov 9, 2016, 16:04

Great!

No, there's no way.. The error relative to each variable is different..

The idea is to do an initial checking in order to avoid a lot of checkings each time a variable is used.

So, when there's a missing variable, it should be reported by the initial checking.

If not, then it means that there's something wrong in the initial checking, so it is a way to detect this kind of failures, like this time with the ```SCRATCH_DIR```.

Thanks @etiennesky !

**Comment by kinow** [comment]
In GitLab by @etiennesky on Nov 9, 2016, 16:14

Thanks for a quick turnaround and explanation @ jlope2 , I am fine with this resolution.

**Comment by kinow** [comment]
In GitLab by @ jlope2 on Nov 9, 2016, 17:10

Mentioned in commit 2296551ed274cd624c1f6ea75553bac2039092f1

**Comment by kinow** [comment]
In GitLab by @ jlope2 on Nov 25, 2016, 17:15

Status changed to closed by commit 2296551ed274cd624c1f6ea75553bac2039092f1
